### ISSUE TITLE
Having more models in a viewmodel when using "factories"

### ISSUE THREAD

**Issue by mlarsson** [kmalakoff/knockback#153]
Hey kmalakoff

I'm facing a problem while using your otherwise wonderful framework. In applesVM below I'm using "factories" to automate the creation of a new viewmodel, a AppleViewModel, when a new apple is added to the applesModel.apples Backbone collection. But the new AppleViewModel also needs access to the singleton instance of my pearModel. But I can't figure out how to inject this model into the new instances of my AppleViewModel. How do I do this in Knockback?

Kind regards Morten Larsson

```
var AppleModel = Backbone.RelationalModel.extend({});

var PearModel = Backbone.RelationalModel.extend({});


var ApplesModel = Backbone.RelationalModel.extend({
    relations: [{
            type: Backbone.HasMany,
            key: "apples",
            relatedModel: AppleModel
        }
    ]
});

var AppleViewModel = kb.ViewModel.extend({
    constructor: function (appleModel, options) {
        //How do I get access to pearModel here as well???
    }
});

var applesVM = kb.ViewModel.extend({
    constructor: function (applesModel, pearModel) {
        var self = this;
        self.apples = kb.collectionObservable(applesModel.get("apples"), {
            factories: {
                "models": AppleViewModel
            }
        });
    }
});
```

**Comment by kmalakoff** [comment]
Glad to hear you've enjoying it. 

Hmmmm. If there was a single model with both apples and a pear, and you configured the reverse relationships for each, and you could traverse the model relationships from apple to apples then to pear when passed in.

```
var ApplesModel = Backbone.RelationalModel.extend({
    relations: [{
            type: Backbone.HasMany,
            key: "apples",
            relatedModel: AppleModel
        }, 
        {
            type: Backbone.HasOne,
            key: "pear",
            relatedModel: PearModel
        }
    ]
});

// plus add the reverse relationships between Apples, Apple, and Pear.
```

Does that work?

**Comment by mlarsson** [comment]
Thank you for your quick response. Yes that would work but it would create an unwanted relation between the models. Apple(s) and the Pear model I mean. They don't really relate to each other. What I'm search for is for the  __viewmodel__ (AppleViewModel) to have access to the PearModel.

**Comment by kmalakoff** [comment]
Most of the time the combination of model relationships or traversing the $data stack in the view using Knockout cover most cases.

I'm trying to think if there is a work around to pass the data through and the best I can think of is using a constructor for the view models. See create API examples like [this](https://github.com/kmalakoff/knockback/blob/master/test/spec/core/view-model.tests.coffee#L290) and [this](https://github.com/kmalakoff/knockback/blob/master/test/spec/core/collection-observable.tests.coffee#L421). If you need this also for the model (non-collection) case, you can also look at the view model tests for create examples.

Basically, you have the option to use a create function install of passing a reference to the view model constructor in the factory. Then you can create an instance of your view model and pass whatever parameters you want to the constructor.

Would this work for you?

**Comment by mlarsson** [comment]
Yes that is exactly what I was looking for! Thank you.
